policyengine_us/variables/gov/states/ri/tax/income/agi/subtractions/taxable_retirement_income/ri_retirement_income_subtraction_eligible.py

```diff
@@ -0,0 +1,36 @@
+from policyengine_us.model_api import *
+
+
+class ri_retirement_income_subtraction_eligible(Variable):
+    value_type = bool
+    entity = TaxUnit
+    label = (
+        "Eligible for the Rhode Island taxable retirement income subtraction"
+    )
+    definition_period = YEAR
+    reference = "http://webserver.rilin.state.ri.us/Statutes/title44/44-30/44-30-12.HTM"
+    # Eligibility is the same as Social Security's eligibility
+    # https://tax.ri.gov/sites/g/files/xkgbur541/files/2022-12/Social%20Security%20Worksheet_w.pdf
+    # STEP 1: Eligibility
+    defined_for = StateCode.RI
+
+    def formula(tax_unit, period, parameters):
+        income = tax_unit("adjusted_gross_income", period)
+        filing_status = tax_unit("filing_status", period)
+        person = tax_unit.members
+        head_or_spouse = tax_unit.any(
```

Comment:
same as above - tax_unit.any shouldn't be here. this will always return true. add tests checking for correct behavior

Reply:
yes agreed, tried to circumvent the `older_spouse_birth_year` being tax_unit but failed to think about the bigger picture logic, will adjust